### PR TITLE
fix: fix evm config init

### PR DESCRIPTION
**Pull request by driftluo**
## What this PR does / why we need it?

This PR fixes evm config init with hardfork features

### What is the impact of this PR?

No Breaking Change

**PR relation**:
- Ref #

<details><summary>CI Settings</summary><br/>

### **CI Usage**

**Tip**: Check the CI you want to run below, and then comment `/run-ci`.

**CI Switch**

- [x] Web3 Compatible Tests
- [ ] OCT 1-5 And 12-15
- [ ] OCT 6-10
- [ ] OCT 11
- [ ] OCT 16-19
- [ ] v3 Core Tests

### **CI Description**

| CI Name                                   | Description                                                               |
| ----------------------------------------- | ------------------------------------------------------------------------- |
| *Web3 Compatible Test*                    | Test the Web3 compatibility of Axon                                       |
| *v3 Core Test*                            | Run the compatibility tests provided by Uniswap V3                        |
| *OCT 1-5 \| 6-10 \| 11 \| 12-15 \| 16-19* | Run the compatibility tests provided by OpenZeppelin                      |

</details>

### CODE REVIEW

**Review by KaoImin — DISMISSED** [review]

**Comment by sunchengzhu** [comment]
This commit (371579d) fixed the issue where `max_contract_limit` was not taking effect when starting the axon node with `hardforks = []`. However, the problem still persists when starting with `hardforks = ["None"]`. Please check. @driftluo 
image

**Comment by sunchengzhu** [comment]
This commit (17c7a66) has been successfully verified locally. @driftluo @KaoImin 
Regardless of starting the Axon node with the configuration `hardforks = [] `or `hardforks = ["None"]`, attempting to deploy a contract that exceeds the max_contract_limit size will result in the following error:
```
jsonRpcResponse: {
  jsonrpc: '2.0',
  error: {
    code: -49998,
    message: 'CKB-VM error execution reverted: ',
    data: 'CreateContractLimit'
  },
  id: 2
}

```

**Review by sunchengzhu — DISMISSED** [review]

**Comment by chaoticlonghair** [comment]
> ```
> jsonRpcResponse: {
>   jsonrpc: '2.0',
>   error: {
>     code: -49998,
>     message: 'CKB-VM error execution reverted: ',
>     data: 'CreateContractLimit'
>   },
>   id: 2
> }
> ```

Has nothing to do with the current PR.
Not issue of current PR, it's an already existed issue.

I have already point that in #1457 but be ingored: Why the error message is related to `CKB-VM`?
I tried some other EVM contracts which must be failed, and they could got the same messages sometimes.

**Comment by driftluo** [comment]
> I have already point that in #1457 but be ingored: Why the error message is related to `CKB-VM`? I tried some other EVM contracts which must be failed, and they could got the same messages sometimes.

https://github.com/axonweb3/axon/blob/main/core/api/src/jsonrpc/error.rs#L57
It should be a display problem, this is the display setting returned by rpc

**Comment by KaoImin** [comment]
> > I have already point that in #1457 but be ingored: Why the error message is related to `CKB-VM`? I tried some other EVM contracts which must be failed, and they could got the same messages sometimes.
> 
> [`main`/core/api/src/jsonrpc/error.rs#L57](https://github.com/axonweb3/axon/blob/main/core/api/src/jsonrpc/error.rs?rgh-link-date=2023-10-18T06%3A34%3A36Z#L57) It should be a display problem, this is the display setting returned by rpc

https://github.com/axonweb3/axon/blob/a2d8e04c7627b9590bc8786b936ae54a60004ba2/core/api/src/jsonrpc/error.rs#L57-L58
The error display should be changed as following:

```rust
    #[display(fmt = "EVM error {}", "decode_revert_msg(&_0.ret)")] 
    EVM(TxResp), 
```

File: core/api/src/jsonrpc/error.rs [hunk 1]
```diff
@@ -54,7 +54,7 @@ pub enum RpcError {
     #[display(fmt = "Invalid filter id {}", _0)]
     CannotFindFilterId(u64),
 
-    #[display(fmt = "CKB-VM error {}", "decode_revert_msg(&_0.ret)")]
+    #[display(fmt = "EVM error {}", "decode_revert_msg(&_0.ret)")]
     VM(TxResp),
```
Comment (KaoImin) [inline core/api/src/jsonrpc/error.rs]:
```suggestion
    EVM(TxResp),
```

Comment (KaoImin) [reply core/api/src/jsonrpc/error.rs]: Perhaps a `CkbVM` error will be added later.

**Review by KaoImin — DISMISSED** [review]

**Review by KaoImin — APPROVED** [review]

**Comment by sunchengzhu** [comment]
The error message returned by commit (fe64e40) is as follows:
```
jsonRpcResponse: {
  jsonrpc: '2.0',
  error: {
    code: -49998,
    message: 'EVM error execution reverted: ',
    data: 'CreateContractLimit'
  },
  id: 2
}
```

**Review by sunchengzhu — APPROVED** [review]

File: core/executor/src/lib.rs [hunk 2]
```diff
@@ -324,13 +324,14 @@ impl AxonExecutor {
...
-            let enable_contract_limit_flag = H256::from_low_u64_be(HardforkName::Andromeda as u64);
+            let enable_contract_limit_flag =
+                H256::from_low_u64_be((HardforkName::Andromeda as u64).to_be());
             if latest_hardfork_info & &enable_contract_limit_flag == enable_contract_limit_flag {
                 let handle = MetadataHandle::new(CURRENT_METADATA_ROOT.with(|r| *r.borrow()));
                 let config = handle.get_consensus_config().unwrap();
                 Some(config.max_contract_limit as usize)
             } else {
-                None
+                Some(0x6000)
```
Comment (Flouse) [inline core/executor/src/lib.rs:334]: Is it better to use fn `default_max_contract_limit` instead of hardcoding 0x6000?

**Pull request merged**